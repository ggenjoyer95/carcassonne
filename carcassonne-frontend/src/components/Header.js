import React from "react";
import { Link, useLocation } from "react-router-dom";
function Header() {
  const location = useLocation();

  if (location.pathname.startsWith("/game/")) {
    return null;
  }
  return (
    <header
      style={{
        backgroundColor: "#f5f5f5",
        padding: "5px",
        textAlign: "center",
      }}
    >
      <h1>
        <Link to="/" style={{ textDecoration: "none", color: "#333" }}>
          Carcassonne
        </Link>
      </h1>
    </header>
  );
}

export default Header;
