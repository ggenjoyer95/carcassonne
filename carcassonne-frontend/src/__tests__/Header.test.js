import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Header from "../components/Header";

test("renders Header component without crashing", () => {
  render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );

  const headerElement = screen.getByText(/Carcassonne/i);
  expect(headerElement).toBeInTheDocument();
});

test("Header matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
