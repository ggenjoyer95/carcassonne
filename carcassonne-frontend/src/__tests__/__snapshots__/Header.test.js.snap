// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <header
    style="background-color: rgb(245, 245, 245); padding: 5px; text-align: center;"
  >
    <h1>
      <a
        href="/"
        style="text-decoration: none; color: rgb(51, 51, 51);"
      >
        Carcassonne
      </a>
    </h1>
  </header>
</DocumentFragment>
`;
